### ISSUE TITLE
Update sunspot to Solr 7+?

### ISSUE THREAD

**Issue by gr8bit** [sunspot/sunspot#989]
This is a request for comments :) @mlh758

Since Solr 7 was released 3 years ago I was thinking if you'd be interested in a PR that cleans up sunspot to support the Solr 7+ fields/syntax/features only, maybe for a 3.0 release? We are about to introduce solr as a base stack feature for our projects so we have some time to update it regularly. :)

**Comment by Stex** [comment]
I'd love to see sunspot updated to solr 7. 

I'm currently upgrading our application to use a dockerized version of it and had to alter solr's default schema quite a bit as there were name clashes with the (dynamic) fields sunspot expects. Also, sunspot uses several types that are marked as deprecated which cannot simply be replaced with the new type as e.g. "sort by geodist" doesn't work any more with them.

I wrote a ruby script to apply the necessary changes via solr's schema API which at least makes it manageable.

**Comment by mlh758** [comment]
Apparently I missed this original post, sorry about that! There is a [sample config](https://github.com/sunspot/sunspot/tree/master/examples/solr7_core) in this repo that can get people up and running on 7.

My team is currently using Solr 7 in production on several applications using this library. We _do_ have the deprecation warnings about things like Trie types but we don't use them so it hasn't really come up as a problem. Removing those types would definitely require a v3 of this library though since I believe we would have to also delete some code in the Ruby library to make that work.

**Comment by Stex** [comment]
@mlh758 You're right, sunspot in its current form absolutely works with Solr 7 using the custom schema. I went through the schema API though as having a static `schema.xml` seems to be discouraged in the Solr documentation and I wanted to be able to apply the changes to any newer Solr version.

However, sunspot expects certain dynamic field namings that clash with Solr's base definition, e.g. `*_ds` (date**s** in base solr, date **stored** in sunspot). It would be great to resolve these clashes in a future version of Sunspot.

**Comment by gr8bit** [comment]
I'm currently on that subject, PR soon. Changes made/to be done:
- ~~small local dockerized test setup~~
- use default `managed-schema` and `solrconfig.xml` from Solr 8 as base for recent configuration
- ~~rename dynamicField extensions (e.g. `_e` for Double to `_d`)~~
- ~~remove Trie types~~
- ~~migrate spatial queries from `LatLonType` to `LatLonPointSpatialField`~~
- get all tests in sunspot folder to run
- add new types (UUID, Currency) with tests

**Comment by gr8bit** [comment]
see #993
